modules/vba_documents/app/sidekiq/vba_documents/upload_processor.rb

```diff
@@ -48,10 +48,16 @@ def download_and_process
 
       begin
         @upload.update(metadata: @upload.metadata.merge(original_file_metadata(tempfile)))
-
         validate_payload_size(tempfile)
 
+        # parse out multipart consumer supplied file into individual parts
         parts = VBADocuments::MultipartParser.parse(tempfile.path)
+
+        # Attempt to use consumer supplied file number field to look up the claiments ICN
+        # asap for tracking consumer's impacted
+        icn = find_icn(parts)
+        @upload.update(metadata: @upload.metadata.merge({ 'ICN' => icn })) if icn.present?
```

Comment:
I'd like to suggest calling the metadata field `icn` rather than `ICN` since our naming convention for metadata keys is lower snake case.